source/postprocess/particles.cc

```diff
@@ -102,15 +131,49 @@ namespace aspect
             patches[i].vertices[0] = particle->get_location();
             patches[i].patch_index = i;
             patches[i].n_subdivisions = 1;
-            patches[i].data.reinit(property_information.n_components()+1,1);
+            patches[i].data.reinit(dataset_names.size(),1);
 
             patches[i].data(0,0) = particle->get_id();
 
             if (particle->has_properties())
               {
                 const ArrayView<const double> properties = particle->get_properties();
+
+                unsigned int output_index = 1;
+                unsigned int field_index = 0;
+                unsigned int do_not_increase_for = 0;
+                std::string field_name = "";
+
                 for (unsigned int property_index = 0; property_index < properties.size(); ++property_index)
-                  patches[i].data(property_index+1,0) = properties[property_index];
+                  {
+                    if (do_not_increase_for > 0)
```

Comment:
This may work, but I find the logic very hard to follow, keeping track of all of these different indices is error prone, and doing the string searches of which properties to exclude twice also seems unnecessary. What about the following modification: You could create a `std::vector<unsigned int> property_index_to_output_index`, and fill it in the first loop above (lines 96-101). Each entry in the vector is either 0 (if the property should not be written into `patches`) or an index that replaces `output_index` below in line 174. That way you only need to fill this vector once and can use it without additional search when copying the properties. The `if` in line 172 would then be replaced by `if (property_index_to_output_index[property_index] != 0) then copy properties[property_index] into patches[property_index_to_output_index[property_index]]; else do nothing`. You might need to set several entries in the vector to 0 if one excluded field has several components, but that should be fine. Do you think that would work and be better?